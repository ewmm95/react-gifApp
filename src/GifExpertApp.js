import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ deffaultCategories = [] }) => {
    const [categories, setCategories] = useState(deffaultCategories)
    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>
            <AddCategory
                setCategories={setCategories}
            />
            <ol>
                {
                    categories.map(
                        (category, i) =>
                            <GifGrid
                                key={category}
                                category={category}
                            />
                    )
                }
            </ol>
        </>
    )
}
