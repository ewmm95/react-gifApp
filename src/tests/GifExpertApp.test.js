import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas de <GifExpertApp/>', () => {
    test('Debe comparar con snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar lista de categorias', () => {
        const props = {
            deffaultCategories: ['One Punch', 'HunterXHunter'],
        };
        const wrapper = shallow(<GifExpertApp {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(props.deffaultCategories.length);
    })
    
})
