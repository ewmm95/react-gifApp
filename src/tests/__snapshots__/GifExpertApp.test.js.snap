// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp/> Debe comparar con snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas de <GifExpertApp/> Debe mostrar lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="HunterXHunter"
      key="HunterXHunter"
    />
  </ol>
</Fragment>
`;
